### PR TITLE
initial poking around to get 2.0 to work

### PR DESCRIPTION
**Pull request by hipsterusername**
I've started poking around major areas where the 2.0 repo has variances with our (customized) versions of their code. I've tried to address these one by one as I've run into them. There's a large chunk of Embedding Mgr code in the FrozenCLIPEmbedder class (in modules.py) - 

I copied this over to the new Frozen**Open**CLIPEmbedder class, but it's not quite as simple as a lift and shift. 
latest error happens here:
```
\invokeai\ldm\modules\encoders\modules.py", line 478, in __init__
    self.transformer = C
```

### CODE REVIEW

File: ldm/modules/attention.py [hunk 1]
```diff
@@ -343,40 +300,60 @@ class SpatialTransformer(nn.Module):
...
+            x = self.proj_in(x)
+        for i, block in enumerate(self.transformer_blocks):
+            x = block(x, context=context[i])
+        if self.use_linear:
+            x = self.proj_out(x)
+        x = rearrange(x, 'b (h w) c -> b c h w', h=h, w=w).contiguous()
+        if not self.use_linear:
+            x = self.proj_out(x)
         return x + x_in
+
```
Comment (hipsterusername) [inline ldm/modules/attention.py:359]: I dropped the entire attention.py file from the 2.0 repo to get through a set of errors here.

Clearly there's some surgery to do to bring variances back in where needed

**Comment by hipsterusername** [comment]
@lstein https://discord.com/channels/1020123559063990373/1045157938106482709/1045210094570307615

**Comment by victorca25** [comment]
Might be good to add or mention the ```open_clip``` dependency. Did you use a specific commit? Like ```bb6e834e9c70d9c27d0dc3ecedeebeaeb1ffad6b```?

**Comment by hipsterusername** [comment]
> Might be good to add or mention the `open_clip` dependency. Did you use a specific commit? Like `bb6e834e9c70d9c27d0dc3ecedeebeaeb1ffad6b`?

that’s fair- https://github.com/Stability-AI/stablediffusion/blob/main/environment.yaml is the environment file from 2.0

**Comment by lstein** [comment]
@tildebyte  note the new dependency needed to support SD 2.0!

**Comment by lstein** [comment]
> @lstein https://discord.com/channels/1020123559063990373/1045157938106482709/1045210094570307615

Yes. I saw the note. I think @damian0815 is on the case? I'm not going to be able to do any substantial work until Friday.

**Comment by damian0815** [comment]
i am, in fact, not on the case.

**Comment by lstein** [comment]
I regret the error. If embeddings for the 1.5 and 1.4 models are not broken, then we have a little time to provide support for 2.0. Hopefully it isn't too much effort.

**Comment by hipsterusername** [comment]
My proposal right now is sprint to get the installer fixed, our RC for 2.2 cut, and ship 2.2. 

As the team orients and begins executing on Nodes, we can iron out SD2.0 support (with an eye towards how those changes will map to Nodes work as well), and get a 2.3 version out for SD2.0 support.

**Comment by lstein** [comment]
> My proposal right now is sprint to get the installer fixed, our RC for 2.2 cut, and ship 2.2.
> 
> As the team orients and begins executing on Nodes, we can iron out SD2.0 support (with an eye towards how those changes will map to Nodes work as well), and get a 2.3 version out for SD2.0 support.

Sounds good to me. I'm putting installer through its paces. Looks good so far.

**Comment by thinkyhead** [comment]
Adding in my 2¢ — I would prefer to be able to use a single installation / working copy of the stable-diffusion repo (from Stability-AI) as the basis for both InvokeAI and the AUTOMATIC1111 WebUI. Is that something that is being considered?

**Comment by ausbitbank** [comment]
Neither auto or invoke is a compatible "copy" of the stable diffusion repo, both are their own beasts now. *If* what you really mean is to not have to store 2 copies of the large .ckpt model files in both auto and invoke, look into "symbolic links" to save storage space :)

> Adding in my 2¢ — I would prefer to be able to use a single installation / working copy of the stable-diffusion repo (from Stability-AI) as the basis for both InvokeAI and the AUTOMATIC1111 WebUI. Is that something that is being considered?

**Comment by lstein** [comment]
Cleaning up - I think this PR has been superseded.

**Pull request closed without merging**